js/main.js

```diff
@@ -248,38 +295,152 @@ xhr2.addEventListener('load', function () {
     var mapContainer = document.createElement('div');
     mapContainer.className = 'map-container';
 
+    var starButton = document.createElement('button');
+    starButton.className = 'star-button';
+
+    var isFavorite = favorites.some(function (favorite) {
+      return favorite.displayName === mapName;
+    });
+
+    if (isFavorite) {
+      starButton.classList.add('favorite');
+      mapContainer.style.order = -1;
+    }
+
+    starButton.addEventListener('click', function () {
+      var favoriteIndex = favorites.findIndex(function (favorite) {
+        return favorite.displayName === mapName;
+      });
+
+      if (favoriteIndex === -1) {
+        favorites.unshift(map);
+        this.classList.add('favorite');
+        mapContainer.style.order = -1;
+      } else {
+        favorites.splice(favoriteIndex, 1);
+        this.classList.remove('favorite');
+        mapContainer.style.order = '';
+      }
+
+      localStorage.setItem('favorites', JSON.stringify(favorites));
```

Comment:
You shouldn't be saving to localStorage here, you should be saving to your data model, and then that would be saved to localStorage in `data.js`